bust-a-move-framework/src/main/java/nl/tudelft/bust_a_move20162017/bust_a_move_framework/bubble/powerup/PowerUp.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * File: PowerUp.java
+ *
+ * Class: PowerUp
+ *
+ * Version: 0.0.1
+ *
+ * Date: September 28th, 2016
+ *
+ */
+
+
+package nl.tudelft.bust_a_move20162017.bust_a_move_framework.bubble;
+
+import org.newdawn.slick.Color;
+import org.newdawn.slick.Graphics;
+import org.newdawn.slick.geom.Circle;
+
+/**
+ * PowerUp provides an abstract representation of a PowerUp
+ * that can be recursively applied to a Bubble.
+ *
+ * @author Calvin Nhieu
+ *
+ */
+public abstract class PowerUp extends Bubble {
+  /**
+   * Root bubble component.
+   */
+  protected Bubble rootBubble;
+
+  /**
+   * Empty constructor per abstract specification.
+   *
+   */
+  public PowerUp() {}
+
+  /**
+   * Creates PowerUp instance.
+   * @param bubble  wrapped Bubble component.
+   */
+  public PowerUp(Bubble bubble) {
+    this.bubble = bubble;
+    this.rootBubble = this.getRootBubble();
+  }
+
+  /**
+   * Draws the PowerUp'd Bubble.
+   * @param g  Java Graphics instance
+   */
+  public abstract void draw(final Graphics g);
+
+  /**
+   * Updates the Bubble's position per game tick.
+   */
+  public final void move() {
+    this.rootBubble.move();
```

Comment:
They are changed. 

I wrote the PowerUp classes such that we don't need to keep duplicate references to the root bubble. In order to use a PowerUp Bubble (e.g. RowBomb wrapping a Bubble) in place of a regular Bubble, the overridden methods in a PowerUp reference the rootBubble.